File: 백준/Silver/28278. 스택 2/스택 2.py
import sys

N = int(sys.stdin.readline())
stack = []

for _ in range(N):
    comd = list(map(int, sys.stdin.readline().split()))

    if comd[0] == 1:
        stack.append(comd[1])
    elif comd[0] == 2:
        print(-1 if not stack else stack.pop())
    elif comd[0] == 3:
        print(len(stack))
    elif comd[0] == 4:
        print(1 if not stack else 0)
    elif comd[0] == 5:
        print(stack[-1] if stack else -1)

File: 백준/Silver/28278. 스택 2/스택 2.java
import java.io.*;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int count = Integer.parseInt(bf.readLine());
        Stack<Integer> stack = new Stack<>();
        while(count --> 0){
            String input = bf.readLine();
            StringTokenizer sT = new StringTokenizer(input);
            int num = Integer.parseInt(sT.nextToken());
            boolean isEmpty = stack.isEmpty();
            if(sT.hasMoreTokens())
                stack.push(Integer.parseInt(sT.nextToken()));
            else if (num == 2)
                sb.append(isEmpty? -1 : stack.pop()).append("\n");
            else if (num == 3)
                sb.append(stack.size()).append("\n");
            else if (num == 4)
                sb.append((isEmpty? 1 : 0)).append("\n");
            else if (num == 5)
                sb.append(isEmpty? -1 : stack.peek()).append("\n");
        }
        System.out.println(sb);
    }
}
